orderer/consensus/etcdraft/chain.go

```diff
@@ -1235,6 +1238,48 @@ func pemToDER(pemBytes []byte, id uint64, certType string, logger *flogging.Fabr
 	return bl.Bytes, nil
 }
 
+// isConsensusUpgrade returns true if config block has a consensustype upgrade to smartbft with empty metadata.
+// in all other cases it returns false
+func (c *Chain) isConsensusUpgrade(block *common.Block) bool {
+	ordererConfigFromBlock := func(block *common.Block) (channelconfig.Orderer, bool) {
+		env := &common.Envelope{}
+		if err := proto.Unmarshal(block.Data.Data[0], env); err != nil {
+			return nil, false
+		}
+
+		bundle, err := channelconfig.NewBundleFromEnvelope(env, c.CryptoProvider)
+		if err != nil {
+			return nil, false
+		}
+		nextOrdererConfig, ok := bundle.OrdererConfig()
+		if ok {
+			return nextOrdererConfig, true
+		}
+		return nil, false
+	}
+
+	lastConfigBlock, err := lastConfigBlockFromSupport(c.support)
+	if err != nil {
+		return false
+	}
+
+	oldOrdererConfig, ok := ordererConfigFromBlock(lastConfigBlock)
+	if !ok {
+		return false
+	}
+
+	nextOrdererConfig, ok := ordererConfigFromBlock(block)
+	if !ok {
+		return false
+	}
+
+	transitionTobft, err := c.validateTransitionToSmartbft(oldOrdererConfig, nextOrdererConfig)
+	if transitionTobft && err == nil {
+		return true
+	}
+	return false
```

Comment:
if the error is not nil, how did this get past the pre-consensus checks?

Reply:
it is a redundant check, will remove it.